### PR TITLE
feat: let consumeRun throw to let message broker handle failures

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
Change behavior of computation consumer to let it throw so that message broker can handle failures

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/com/powsybl/ws/commons/computation/ComputationException.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.ws.commons.computation;
+
+/**
+ * @author Joris Mancini <joris.mancini_externe at rte-france.com>
+ */
+public class ComputationException extends RuntimeException {
+    public ComputationException(String message, Throwable cause) {
```
Comment (jonenst) [inline src/main/java/com/powsybl/ws/commons/computation/ComputationException.java:13]: probably better to add the constructor without cause right now

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java [hunk 2]
```diff
@@ -94,24 +92,26 @@ protected void cleanResultsAndPublishCancel(UUID resultUuid, String receiver) {
...
         try {
             cancelComputationRequests.put(cancelContext.resultUuid(), cancelContext);
 
             // find the completableFuture associated with result uuid
             CompletableFuture<R> future = futures.get(cancelContext.resultUuid());
-            if (future != null) {
-                isCanceled = future.cancel(true);  // cancel computation in progress
-                if (isCanceled) {
+            if (future == null) {
+                cleanResultsAndPublishCancel(cancelContext.resultUuid(), cancelContext.receiver());
```
Comment (jonenst) [inline src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:103]: this will be run on all but one server ?

Comment (jonenst) [reply src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:103]: it looks like it's already the case after this method returns...?

Comment (jonenst) [reply src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:103]: before we did publishCancelFail but now we do cleanResults and publishStop

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java [hunk 3]
```diff
@@ -94,24 +92,26 @@ protected void cleanResultsAndPublishCancel(UUID resultUuid, String receiver) {
...
+            } else {
+                boolean isCanceled = future.cancel(true);  // cancel computation in progress
+                if (future.isDone() || isCanceled) {
```
Comment (jonenst) [inline src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:106]: Not sure about adding this clean when the computation is done

File: src/main/java/com/powsybl/ws/commons/computation/ComputationException.java (hunk 1 above)
Comment (TheMaskedTurtle) [reply src/main/java/com/powsybl/ws/commons/computation/ComputationException.java:17]: done

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java (hunk 2 above)
Comment (TheMaskedTurtle) [reply src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java]: removed

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java [hunk 4]
```diff
@@ -143,15 +141,12 @@ public Consumer<Message<String>> consumeRun() {
                     sendResultMessage(resultContext, result);
                     LOGGER.info("{} complete (resultUuid='{}')", getComputationType(), resultContext.getResultUuid());
                 }
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
+            } catch (CancellationException e) {
+                LOGGER.info("Computation was interrupted");
```
Comment (jonenst) [inline src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:145]: Better to add context (ids or something) to this log just like above it ?

Comment (antoinebhs) [reply src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:145]: this is probably a redundant log? we already log the cancellation elsewhere:
```
c.p.w.c.c.service.AbstractWorkerService  : Computation was interrupted
....
....
....
c.p.w.c.c.service.AbstractWorkerService  : Security analysis was canceled (resultUuid='49409bd8-59d5-45ef-adcc-ff8293f4dc25')
c.p.o.sa.AbstractSecurityAnalysis        : Security analysis cancelled in 8321 ms
```

File: src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java [hunk 5]
```diff
@@ -143,15 +141,12 @@ public Consumer<Message<String>> consumeRun() {
                     sendResultMessage(resultContext, result);
                     LOGGER.info("{} complete (resultUuid='{}')", getComputationType(), resultContext.getResultUuid());
                 }
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
+            } catch (CancellationException e) {
```
Comment (antoinebhs) [inline src/main/java/com/powsybl/ws/commons/computation/service/AbstractWorkerService.java:144]: Not sure why it's not covered by tests

File: src/main/java/com/powsybl/ws/commons/computation/ComputationException.java (hunk 1 above)
Comment (TheMaskedTurtle) [reply src/main/java/com/powsybl/ws/commons/computation/ComputationException.java:17]: drawback of adding unused code, it's not covered by test ^^

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**